src/net_processing.cpp

```diff
@@ -1062,23 +1059,22 @@ unsigned int LimitOrphanTxSize(unsigned int nMaxOrphans)
  * Increment peer's misbehavior score. If the new value >= DISCOURAGEMENT_THRESHOLD, mark the node
  * to be discouraged, meaning the peer might be disconnected and added to the discouragement filter.
  */
-void Misbehaving(NodeId pnode, int howmuch, const std::string& message) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+void Misbehaving(const NodeId pnode, const int howmuch, const std::string& message) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
 {
-    if (howmuch == 0)
-        return;
+    assert(howmuch > 0);
```

Comment:
I've verified all callers, this is is correct.

Reply:
> I've verified all callers, this is is correct.
While this is true,
if you assert on a const that gives the cpu regs a lot of control power, imho.

Reply:
@saibato I have no clue what you're trying to say here.

Reply:
we changed  howmuch to const int , and then we assert, i saw such things optimized to regs. and the cpu is not always in your control and her we are at a net sanity check that should not be hookable.imho.

Reply:
@saibato Please stop spewing gibberish.